### ISSUE TITLE
Access to `->body()` while building custom cache key at cacheKey()

### ISSUE THREAD

**Issue by boryn** [Sammyjo20/Saloon#193]
As some endpoints pass the parameters through the body using POST method (and some through query using GET), I wanted to specifically build the cache key using this information as well.

The `CacheKeyHelper` already uses `$query = $pendingRequest->query()->all();` and in my own `cacheKey(PendingRequest $pendingRequest)` implementation, I wanted to similarly use: `$body = $pendingRequest->body()->all();`. Unfortunately the `body()` is not accesible there, I get `Call to a member function all() on null`.

**Comment by Sammyjo20** [comment]
Hey @boryn 

Ah I have found a way we can get around this - I will just implement a fix for you.

**Comment by Sammyjo20** [comment]
This has now been fixed in v2.1 of the cache plugin 🔥 Thank you for raising this! https://github.com/Sammyjo20/saloon-cache-plugin/releases/tag/v2.1.0

**Comment by boryn** [comment]
Hi @Sammyjo20 !

Looking at the original `CacheKeyHelper`:

```php
public static function create(PendingRequest $pendingRequest): string
{
    $className = $pendingRequest->getRequest()::class;
    $requestUrl = $pendingRequest->getUrl();
    $query = $pendingRequest->query()->all();
    $headers = $pendingRequest->headers()->all();

    return json_encode(compact('className', 'requestUrl', 'query', 'headers'), JSON_THROW_ON_ERROR);
}
```

I started to think if it may happen that `$pendingRequest->query()` or `$pendingRequest->headers()` is null? It's possible with `$pendingRequest->body()` that this is null.

In that case not to raise an error of running `->all()` on `null` we'd rather need: 
```php
$query = optional($pendingRequest->query())->all();
$headers = optional($pendingRequest->headers())->all();
```
